server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -337,6 +345,16 @@ func (e *EventChannel) handleEvent(event *pepb.PublishBuildToolEventStreamReques
 func (e *EventChannel) processSingleEvent(event *inpb.InvocationEvent, iid string) error {
 	e.beValues.AddEvent(event.BuildEvent) // in-memory structure to hold common values we want from the event.
 
+	switch p := event.BuildEvent.Payload.(type) {
+	case *build_event_stream.BuildEvent_Progress:
+		if e.env.GetConfigurator().GetStorageEnableChunkedEventLogs() {
+			e.logWriter.Write([]byte(p.Progress.Stderr))
+			e.logWriter.Write([]byte(p.Progress.Stdout))
+			p.Progress.Stderr = ""
+			p.Progress.Stdout = ""
+		}
```

Comment:
Where do we write to the screen writer / terminal thingy currently? Could we write to the chunkstore in that location too or is there a good reason to have this in a different location?

Reply:
The place it is currently being written is in the [EventParser](https://github.com/buildbuddy-io/buildbuddy/blob/master/server/build_event_protocol/event_parser/event_parser.go#L142-L148), which is also used to look up invocations, which is not a time we want to be writing chunks. I thought this would be better, since even if we decided to conditionally write chunks, it's still a weird side effect to add to the function we use to parse events.

Reply:
SGTM

Since this mutates the event (p.Progress.Std* = ""), does it get called before / after that code or interfere with it?

Reply:
It gets called before, which prevents the event log from being stored in the invocation if we are using the chunked logs strategy (by design). Does that seem good to you? Are there alternatives that you might consider preferable?

Reply:
(putting this here for completeness) We should continue to write the data in the old-style *as well* as the new style, and then control which style is displayed based on a flag. That way if we need to roll back we can easily do so. Once we are happy that the new-style stuff is 100% awesome, then we can stop writing old-style logs.

Reply:
(also -- rest of this CL LGTM, so once this is fixed i'll approve)

Reply:
Fixed it! Removed the erasure and ensured the config flag will be pushed over the wire to the client